### DISCUSSION TITLE
Detecting Connected Group Resistors

### DISCUSSION THREAD

**Discussion by Abbas111977** [SpiceSharp/SpiceSharp#205] [General]
Hi, I am trying to find all the sub circuits (the resistors that are connected together) inside a bigger circuit , so i came up with the following algorithm:
1. i have entered the hole circuit (which contains resistors "Islands") 
2. i have added source to one of the resistors and assumed that the ground is connected always to single resistor (resistor between 0 and MG_0 nodes)
3. i have connected the +V voltage to one of the resistors with floating nodes (R_*25B10 as shown in the attached pic.)
4. i simulated the circuit and return the floating rule violations as result.
5. depending on the rule violations I have found the resistors "Islands" (the resistors that their nodes doesn't violate the floating rule)
mostly the algorithm worked, but for the attached circuits the nodes MG_7, MG_8 and MG_9 are detected as floating rule violation nodes !!!
Is there is any thing that I am doing wrong ? 
How can I solve this problem ?
thanks for your kind support

![Floating components detection problem](https://user-images.githubusercontent.com/62871714/177559208-8c637fa0-8141-49b5-b2ee-65b550ac5614.jpg)

**Comment by svenboulanger** [comment]
I tried entering this circuit in Spice#.Parser and it seems to work fine?

```
V1 MG_0 0 1
R1 MG_0 0 1k

R2 MG_0 MG_6 1k
R3 T1 MG_6 1k

R4 MG_6 MG_301 1k
R5 MG_301 MG_9 1k
R6 MG_9 T2 1k

R7 MG_6 MG_299 1k
R8 MG_299 MG_8 1k
R9 MG_8 T3 1k

R10 MG_6 MG_297 1k
R11 MG_297 MG_7 1k
R12 T4 MG_7 1k

R13 MG_6 MG_303 1k
R14 MG_303 MG_10 1k
R15 MG_10 T5 1k

.OP
.END
```
Can you maybe supply a netlist or some code that does give an error?

[EDIT] If you remove R1 it does give a rule violation. But I'll have to take a closer look at home why that happens.

**Reply by Abbas111977** [reply]
Hi, thank you for fast reply, i think that you have connected the source in parallel with r1 thats why when you remove it there will be violation

**Reply by Abbas111977** [reply]
Just to let you know that i use the violations as a result to determin which component is not connected, so i dont want to get rid of the violations because i need them to determin the electrically connected parts.

**Reply by svenboulanger** [reply]
It turns out that there isn't a mistake. I just forgot that in Spice netlists, the first line is always ignored as it is the "title line". Having the source on the second line gives no rule violations.

So that means I cannot reproduce the error you are having. Can you please share some simple example or netlist that produces the errors?

**Reply by Abbas111977** [reply]
I think there is misunderstanding, in fact I need the floating violation error list so that I can use it to find the connected resistors network and distinguish them from other connected resistor networks, referring to the attached pic, i need to generate three lists each one contains the resistors that are electrically connected and forming a network i.e.  N0, N1, N2 resistors networks and I am using the floating rule violation to do that
![Error](https://user-images.githubusercontent.com/62871714/177992528-68122b0b-e37f-4aa8-a7e8-a93a8018faba.jpg)
.

**Reply by svenboulanger** [reply]
I think it should work, because I have done it with the library myself. I create an OP simulation, and run it within a try-catch block. Then I catch the `ValidationFailedException` and with the exception you should be able to get to the violated rules.

```csharp
// using System.Linq;

try
{
    op.Run(ckt);
}
catch (ValidationFailedException ex)
{
    // Let's fix floating nodes
    var violations = ex.Rules.Violations.OfType<FloatingNodeRuleViolation>();
    // ...
}
```

The issue is that for your example you mention that
> mostly the algorithm worked, but for the attached circuits the nodes MG_7, MG_8 and MG_9 are detected as floating rule violation nodes !!!

But I cannot reproduce any floating node rule violations for the circuit that you give. Perhaps you are doing something different in your code, or else there might be a bug in Spice# in which case I would like to be able to reproduce the bug.
